test/api/test_jobs.py

```diff
@@ -200,8 +200,8 @@ def __history_with_new_dataset( self ):
         return history_id, dataset_id
 
     def __history_with_ok_dataset( self ):
-        history_id, dataset_id = self.__history_with_new_dataset()
-        self._wait_for_history( history_id, assert_ok=True )
+        history_id = self._new_history()
+        dataset_id = self._new_dataset( history_id, wait=True )[ "id" ]
```

Comment:
It seems that this method is no more checking that the state of the uploaded dataset is "ok", which contradicts the method name.